gulpfile.js

```diff
@@ -0,0 +1,37 @@
+const gulp = require('gulp');
+const gutil = require('gulp-util');
+const typescript = require('gulp-typescript');
+const sourcemaps = require('gulp-sourcemaps');
+const tsProject = typescript.createProject('tsconfig.json');
+
+const OPTIONS = {
+	header: true,
+	files: {
+		typescript: [ 'lib/**/*.ts' ],
+		declarations: [ 'lib/**/*.d.ts' ]
+	},
+	base: 'lib'
+}
+
+// Compile the TS sources
+gulp.task('typescript', () => {
+	gulp.src(OPTIONS.files.typescript)
+		.pipe(sourcemaps.init())
+		.pipe(tsProject()).on('error', gutil.log)
+		.pipe(sourcemaps.write('./', { includeContent: false,
+			sourceRoot: '../lib',
+			// There is currently an issue where not doing this won't generate
+			// correctly pathed mapfiles.
```

Comment:
Is there a github issue for this somewhere? Would be nice to link here, so we can kill this once it's sorted.